src/EventStore.Core/Services/Transport/Grpc/Enumerators.AllSubscription.cs

```diff
@@ -44,26 +42,16 @@ public class AllSubscription : IAsyncEnumerator<ReadResp> {
 				ClaimsPrincipal user,
 				bool requiresLeader,
 				IReadIndex readIndex,
-				ReadReq.Types.Options.Types.UUIDOption uuidOption,
 				CancellationToken cancellationToken) {
-				if (bus == null) {
-					throw new ArgumentNullException(nameof(bus));
-				}
-
-				if (readIndex == null) {
-					throw new ArgumentNullException(nameof(readIndex));
-				}
-
 				_expiryStrategy = expiryStrategy;
 				_subscriptionId = Guid.NewGuid();
-				_bus = bus;
+				_bus = bus ?? throw new ArgumentNullException(nameof(bus));
```

Comment:
i could be persuaded, but i think i'd be tempted to leave these tidyups for another PR, so that this one can be as minimal and obviously correct as possible

Reply:
sure I will update this.